src/app/providers/providers.service.ts

```diff
@@ -37,6 +43,16 @@ export class ProvidersService {
         providerId: project.provider.providerId
       };
       return this.gitlabService.gitlabPost(importData);
+    } else if (project.provider.name === ProviderType.GitHub.toLowerCase()) {
+      const importData: GitHubMainImport = {
+        url: project.provider.url,
+        tokenId: project.provider.tokenId,
+        group: project.groupName,
+        project: project.name,
+        providerId: project.provider.providerId,
+        uploadUrl: project.provider.uploadUrl,
+      };
+      return this.gitHubService.githubPost(importData);
```

Comment:
nit: Suggest changing this to a `switch` statement

Reply:
I wrote this version 
```ts
  refreshProject(project: WharfProject): Observable<any> {
    let importData: GitHubMainImport | GitlabMainImport | AzureDevOpsMainImport;
    switch (project.provider.name) {
      case ProviderType.GitLab.toLowerCase():
        importData = {
          url: project.provider.url,
          tokenId: project.provider.tokenId,
          group: project.groupName,
          project: project.name,
          providerId: project.provider.providerId
        };
        return this.gitlabService.gitlabPost(importData);
      case ProviderType.GitHub.toLowerCase():
        importData = {
          url: project.provider.url,
          tokenId: project.provider.tokenId,
          group: project.groupName,
          project: project.name,
          providerId: project.provider.providerId,
          uploadUrl: project.provider.uploadUrl,
        };
        return this.gitHubService.githubPost(importData);
      case ProviderType.AzureDevOps.toLowerCase():
        importData = {
          url: project.provider.url,
          tokenId: project.provider.tokenId,
          group: project.groupName,
          project: project.name,
          providerId: project.provider.providerId
        };
        return this.azureDevOpsService.azuredevopsPost(importData);
    }
  }
```

Because of the nature of the block the const cant be redeclared though so we would loose the type saftey it seems. I vote it stays as is.

Reply:
Taken! Thanks.